src/lib/encoder.ts

```diff
@@ -325,6 +340,10 @@ const valueTypeEncodingMap = {
     decode: (value: string) => abiCoder.decodeParameter('address', value),
   },
   // NOTE: We could add conditional handling of numeric values here...
+  uint128: {
+    encode: (value: string | number) => uint128ToHex(value),
+    decode: (value: string) => hexToUint128(value),
```

Comment:
I think you can write the functions here directly instead of making the functions hexToUint128 etc.

Reply:
I was about to put a review comment for the same thing.

We can probably extract to a function later, we will have to support multiple uintN